src/audio/eq_iir/eq_iir.c

```diff
@@ -286,27 +312,39 @@ static void eq_iir_s32_24_default(const struct comp_dev *dev, const struct audio
 }
 #endif /* CONFIG_FORMAT_S32LE && CONFIG_FORMAT_S24LE */
 
-static void eq_iir_pass(const struct comp_dev *dev,
-			const struct audio_stream *source,
-			struct audio_stream *sink,
-			uint32_t frames)
+static void eq_iir_pass(struct processing_module *mod, struct input_stream_buffer *bsource,
+			struct output_stream_buffer *bsink, uint32_t frames)
 {
+	struct audio_stream *source = bsource->data;
+	struct audio_stream *sink = bsink->data;
+
+	if (source->frame_fmt == SOF_IPC_FRAME_S16_LE) {
+		bsource->consumed += (frames * source->channels) << 1;
+		bsink->size += (frames * source->channels) << 1;
+	} else {
+		bsource->consumed += (frames * source->channels) << 2;
+		bsink->size += (frames * source->channels) << 2;
+	}
```

Comment:
Maybe all of this could happen in one function since many components have  a pass-thru mode.